### PR TITLE
Chore/update to latest api defs

### PR DESCRIPTION
**Pull request by aarlaud**
- [x] Tests written and linted [ℹ︎](https://github.com/snyk/general/wiki/Tests)
- [ ] Documentation written [ℹ︎](https://github.com/snyk/general/wiki/Documentation)
- [x] Commit history is tidy [ℹ︎](https://github.com/snyk/general/wiki/Git)

### What this does

Updates the current api client to be in sync with latest v1 endpoints.

### CODE REVIEW

File: src/lib/client/abstraction/org/projects.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  maxRetryCount: 10,
+  userAgentPrefix: 'snyk-api-ts-client',
+});
+
+export const getV3 = async (
+  classContext: Object,
+): Promise<ProjectsPostResponseType> => {
+  const response: ProjectsPostResponseType = {};
+  let url = '';
+  let urlQueryParams: Array<string> = ['version=2023-05-29', 'limit=10'];
```
Comment (nathan-roys) [inline src/lib/client/abstraction/org/projects.ts:21]: Is there potentially a reason to allow a customer to specify a version of the API they'd want to use?

File: src/lib/client/abstraction/org/projects.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  try {
+    const resultSet: ProjectsData[][] = [];
+    let isThereNextPage: boolean = false;
+    do {
+      const result = await requestManager.request({
+        verb: 'get',
+        url: url,
+        useRESTApi: true,
+      });
+      isThereNextPage = result.data.links.next ? true : false;
```
Comment (nathan-roys) [inline src/lib/client/abstraction/org/projects.ts:37]: Can this just be:

`isThereNextPage = result.data.links.next`

?

**Review by nathan-roys — APPROVED** [review]
Added a couple of comments, but LGTM

File: src/lib/client/abstraction/org/projects.ts (hunk 1 above)
Comment (aarlaud) [reply src/lib/client/abstraction/org/projects.ts:21]: yes, but I think that would require v3 to be added as client formally, not in the abstraction here.

File: src/lib/client/abstraction/org/projects.ts (hunk 2 above)
Comment (aarlaud) [reply src/lib/client/abstraction/org/projects.ts:37]: no, next is a link, and we need a boolean if next link exists

**Pull request merged**